src/gsy_e/models/market/two_sided.py

```diff
@@ -344,15 +344,17 @@ def match_recommendations(
                     # re-raise exception to be handled by the external matcher
                     raise invalid_bop_exception
                 continue
-            original_bid_rate = market_bid.original_price / market_bid.energy
-            if ConstSettings.MASettings.BID_OFFER_MATCH_TYPE == BidOfferMatchAlgoEnum.PAY_AS_BID:
+            original_bid_rate = recommended_pair.bid_energy_rate + (
```

Comment:
Is this really a `+` ? Shouldn't be the grid_fee rate subtracted from the `bid_energy_rate` in order to get the `original_bid_rate`?

Reply:
In the case of offers, this is true. However, the bid's original rate is higher than the propagated rate, so the original rate is indeed `propagated_rate`+ `fees`

Reply:
Sorry, I do not agree.
Lets say the original bid in the House market has the rate of 30ct/kWh.
If this bid is forwarded to the Community market with 2ct/kWh fee, the propagated bid rate is 32cent/kWh.
Consequently: original_bid_rate (30) = propagated_rate (32) - fee_rate (2)
Or do I get it totally wrong?

Reply:
I think that it's the opposite, in your example the bid in the community market will be 28 cents (The original buyer isn#t willing to pay more than 30).
So the bid will be cleared at 28 cents/kwh, the community area will sell the energy to the bid buyer for 30 cents/kwh (SO its share of 2 cents/kwh is guaranteed)

Reply:
Thanks a lot for bering with me! I had to recapitulate how the grid fees were implemented. You are right of course, the seller should pay the fees and the market is called `pay as bid`, so the bid price should not be exceeded. Thanks again!